runtime/compiler/env/VMJ9.cpp

```diff
@@ -6589,9 +6589,9 @@ TR_J9VMBase::getInstanceFieldOffset(TR_OpaqueClassBlock * clazz, char * fieldNam
 TR_OpaqueClassBlock *
 TR_J9VM::getSuperClass(TR_OpaqueClassBlock * classPointer)
    {
-   J9Class * clazz = TR::Compiler->cls.convertClassOffsetToClassPtr(classPointer);
+   J9Class *clazz = TR::Compiler->cls.convertClassOffsetToClassPtr(classPointer);
    UDATA classDepth = J9CLASS_DEPTH(clazz) - 1;
-   return convertClassPtrToClassOffset(classDepth >= 0 ? clazz->superclasses[classDepth]: 0);
+   return convertClassPtrToClassOffset(clazz->superclasses[classDepth]);
```

Comment:
I believe that `J9CLASS_DEPTH` is zero for `java.lang.Object`, so I'm guessing that this method is never called for that class.  Looking back at the history of this file, it looks like the type of `classDepth` &mdash; until fourteen years ago &mdash; was `int32_t`.  Back then, the check for `classDepth >= 0` made sense.

I would suggest changing this to be safe `java.lang.Object`:

```
   UDATA classDepth = J9CLASS_DEPTH(clazz);
   return convertClassPtrToClassOffset(classDepth >=1 ? clazz->superclasses[classDepth-1] : 0);
```